test/CadlRanchProjectsNonAzure.Tests/type-property-nullable.cs

```diff
@@ -0,0 +1,248 @@
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetDurationClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(XmlConvert.ToTimeSpan("P123DT22H14M12.011S"), response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_getNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetDurationClient().GetNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(null, response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_patchNonNull() => Test(async (host) =>
+        {
+            var value = new
+            {
+                requiredProperty = "foo",
+                nullableProperty = "P123DT22H14M12.011S"
+            };
+            var response = await new NullableClient(host, null).GetDurationClient().PatchNonNullAsync(BinaryContent.Create(BinaryData.FromObjectAsJson(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetDurationClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(BinaryData.FromString("hello, world!").ToString(), response.Value.NullableProperty.First().ToString());
+            Assert.AreEqual(BinaryData.FromString("hello, world!").ToString(), response.Value.NullableProperty.Last().ToString());
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_getNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().GetNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            // we will never construct a null collection therefore this property is actually undefined here.
+            Assert.IsNotNull(response.Value.NullableProperty);
+            Assert.IsFalse(!(response.Value.NullableProperty is ChangeTrackingList<BinaryData> changeTrackingList && changeTrackingList.IsUndefined));
```

Comment:
Could we change this to use the existing method `Optional.IsCollectionDefined()`